resources/lib/common/kodiops.py

```diff
@@ -273,3 +273,12 @@ def is_less_version(self, ver):
 
     def __str__(self):
         return self.build_version
+
+
+def make_legal_filename(filename):
+    """Wrapper for makeLegalFilename"""
+    if hasattr(xbmc, 'makeLegalFilename'):  # Kodi 18.x
```

Comment:
Can you just make this an import in the top level scope for this module?
```
if hasattr(xbmc, 'makeLegalFilename'):
  from xbmc import makeLegalFilename
else:
  from xbmcvfs import makeLegalFilename
```

More typically, I'd see:
```
try:
  from xbmcvfs import makeLegalFilename
except ImportError:
  from xbmc import makeLegalFilename
```

`</nitpick>`

Reply:
I wouldn't know the best way
because before the final version of Kodi 19 there will be others API changes

in this way will lead to many import errors at every execution of the first instance
I know that making a lot of exceptions makes performance worse

Reply:
As long as you aren't generating lots of errors in a loop it's generally fine (imports are a one-time cost). And as people upgrade Kodi they'll stop hitting the fallback branch as well. If you really want to avoid the try/except you can still do the hasattr check, I just figured you could remove the extra function wrapping

Reply:
understood well i will use try except